minio/known_error.py

```diff
@@ -0,0 +1,369 @@
+class MinioError(Exception):
+    """
+    Base class for all exceptions
+
+    :param message: User defined message.
+    """
+    def __init__(self, message, **kwargs):
+        super(MinioError, self).__init__(**kwargs)
+        self.message = message
+
+    def __str__(self):
+        return "{name}: message: {message}".format(
+            name=self.__class__.__name__,
+            message=self.message
+        )
+
+class KnownResponseError(MinioError):
+    def __init__(self, response_error, **kwargs):
+        super(KnownResponseError, self).__init__(message=self.message, **kwargs)
+        self.response_error = response_error
+
+class AccessDenied(KnownResponseError):
+    message = 'Access Denied'
+
+class AccountProblem(KnownResponseError):
+    message = 'There is a problem with your account that prevents the ' \
+              'operation from completing successfully.'
+
+class AmbiguousGrantByEmailAddress(KnownResponseError):
+    message = '	The email address you provided is associated with ' \
+              'more than one account.'
+
+class BadDigest(KnownResponseError):
+    message = 'The Content-MD5 you specified did not match what we received.'
+
+class BucketAlreadyExists(KnownResponseError):
+    message = 'The requested bucket name is not available. The ' \
+              'bucket namespace is shared by all users of the system. ' \
+              'Please select a different name and try again.'
+
+class BucketAlreadyOwnedByYou(KnownResponseError):
+    message = 'Your previous request to create the named bucket ' \
+              'succeeded and you already own it.'
+
+class BucketNotEmpty(KnownResponseError):
+    message = 'The bucket you tried to delete is not empty.'
+
+class CrecentialNotSupported(KnownResponseError):
+    message = '	This request does not support credentials.'
```

Comment:
I honestly don't know why this happens to all of my code. I dev in PyCharm if that makes any difference.